plugins/modules/files/json_patch.py

```diff
@@ -0,0 +1,553 @@
+  json_patch:
+    src: "test.json"
+    backup: yes
+    operations:
+      - op: move
+        from: "/2/baz/2/bar"
+        path: "/0/foo/bar"
+
+- name: Test that the "foo" object has three members
+  json_patch:
+    src: "test.json"
+    operations:
+      - op: test
+        path: "/0/foo/one"
+        value: 1
+      - op: test
+        path: "/0/foo/two"
+        value: 2
+      - op: test
+        path: "/0/foo/three"
+        value: 3
+'''
+
+
+RETURN = r'''
+tested:
+    description: The result of any included test operation.
+    returned: when test operation exists
+    type: bool
+backup:
+    description: The name of the backed up file.
+    returned: when backup is true
+    type: str
+dest:
+    description: The name of the file that was written.
+    returned: changed
+    type: str
+created:
+    description: Whether the file was newly created.
+    returned: always
+    type: bool
+'''
+
+
+import json
+import os
+import tempfile
+
+from ansible.module_utils import basic
+from ansible.module_utils._text import to_bytes, to_native
+
+
+def set_module_args(args):
+    """For dynamic module args (such as for testing)."""
+    args = json.dumps({'ANSIBLE_MODULE_ARGS': args})
+    basic._ANSIBLE_ARGS = to_bytes(args)
+
+
+class PathError(Exception):
+    """Raised when no valid JSON object or property is found for a given path."""
+    pass
+
+
+class PatchManager(object):
+    """Manage the Ansible portion of JSONPatcher."""
+
+    def __init__(self, module):
+        self.module = module
+        self.create = self.module.params.get('create', False)
+        self.create_type = self.module.params.get('create_type', 'object').lower()
+        empty = False
+
+        # validate file
+        self.src = self.module.params['src']
+        if not os.path.isfile(self.src):
+            if not self.create:
+                self.module.fail_json(msg="could not find file at `%s`" % self.src)
+            empty = True
+
+        # use 'src' as the output file, unless 'dest' is provided
+        self.outfile = self.src
+        self.dest = self.module.params.get('dest')
+        if self.dest is not None:
+            self.outfile = self.dest
+
+        try:
+            self.json_doc = open(self.src).read() if not empty else ""
+        except IOError:
+            self.module.fail_json(msg="could not read file at `%s`" % self.src)
+
+        # create empty JSON if requested
+        if self.json_doc == "" and self.create:
+            if self.create_type == "object":
+                self.json_doc = "{}"
+            elif self.create_type == "array":
+                self.json_doc = "[]"
+            else:
+                self.module.fail_json(msg="invalid option for 'create_type': %s" % self.create_type)
+
+        self.operations = self.module.params['operations']
+        try:
+            self.patcher = JSONPatcher(self.json_doc, *self.operations)
+        except Exception as e:
+            self.module.fail_json(msg=str(e))
```

Comment:
```suggestion
            self.module.fail_json(msg=to_native(e))
```
to_native is used, among other things, for such cases, allows to handle output independently from python version